### ISSUE TITLE
Wrong traversal order in Data.IntMap.mapAccumRWithKey 

### ISSUE THREAD

**Issue by sjakobi** [haskell/containers#690]
```
> mapAccumRWithKey (\acc k v -> (acc ++ show k ++ show v, v)) "" (Data.IntMap.fromList [(-1, "a"), (2, "b"),(-3, "c"), (4, "d")])
("-1\"a\"-3\"c\"4\"d\"2\"b\"",fromList [(-3,"c"),(-1,"a"),(2,"b"),(4,"d")])
```

1. Positive keys should come first.
~~2. Negative keys should be traversed in descending order too.~~ (That's already correct)

#689 is related.

**Comment by sjakobi** [comment]
Both the strict and the lazy versions need to be fixed here.

**Comment by Vlix** [comment]
Already mentioned in #689 that this is now fixed in #688

**Comment by sjakobi** [comment]
Fixed in #688.
